R/create_country_codes.R

```diff
@@ -0,0 +1,303 @@
+#' Country Code Lookup
+#'
+#' @description
+#' These pre-defined country codes are sourced from
+#' [ISO 3166 Standards](https://www.iso.org/iso-3166-country-codes.html).
+#' See also [Wikipedia](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3).
+#'
+#' @details
+#'
+#' `country_code` is the 3-letter ISO 3166-1 county code commonly found in the
+#' ADSL `COUNTRY` variable.
+#' `country_name` is the country long name corresponding to to the 3-letter code.
+#' `country_number` is the numeric code corresponding to an alphabetic sorting of
+#' the 3-letter codes.
+#'
+#' To see the entire table in the console, run `print(country_code_lookup)`.
+#'
+#' @seealso [dose_freq_lookup]
+#'
+#' @export
+#'
+#' @keywords metadata
+#'
+#' @family metadata
+#'
+#' @examples
+#' library(tibble)
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(lubridate)
+#'
+#' # Create reference dataset for periods
+#' adsl <- tribble( #' adsl <- tribble(
+#'   ~USUBJID, ~SEX, ~COUNTRY,
+#'   "ST01-01", "F", "AUT",
+#'   "ST01-02", "M", "MWI",
+#'   "ST01-03", "F", "GBR",
+#'   "ST01-04", "M", "CHE",
+#'   "ST01-05", "M", "NOR",
+#'   "ST01-06", "F", "JPN",
+#'   "ST01-07", "F", "USA"
+#' )
+#'
+#' covar <- adsl %>%
+#'   derive_vars_merged(
+#'     dataset_add = country_code_lookup,
+#'     new_vars = exprs(COUNTRYN = country_number, COUNTRYL = country_name),
+#'     by_vars = exprs(COUNTRY = country_code)
+#'   )
+#' covar
+#'
+#' @rdname country_code_lookup
+country_code_lookup <- tribble(
```

Comment:
Could we derive `country_number` instead of hard-coding it? Then it is easier to maintain.

Reply:
Updated, thanks!